### PR TITLE
R doc 1761 update subscriptions

### PR DESCRIPTION
**Pull request by morhilai**
Documented new method Subscriptions.Update() / UpdateAsync()

### CODE REVIEW

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.dotnet.markdown [hunk 1]
```diff
@@ -70,7 +70,12 @@ usage:
 {PANEL/}
 
 {PANEL: Updating subscription}
-{INFO This operation can only be performed through the management studio /}
+
+See [example](../../../client-api/data-subscriptions/creation/examples#update-existing-subscription) 
+and [api description](../../../client-api/data-subscriptions/creation/api-overview#update-subscription).
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.dotnet.markdown:75]: api -> API

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown [hunk 2]
```diff
@@ -50,18 +52,51 @@ Non generic version of the class, relies on user's full knowledge of the RQL que
...
 
+{CODE sub_create_options_strong@ClientApi\DataSubscriptions\DataSubscriptions.cs /}
+
 | Member | Type | Description |
 |--------|:-----|-------------| 
 | **&lt;T&gt;** | type | Type of the object, from which the collection will be derived. |
 | **Name** | `string` | User defined name of the subscription: allows to have a human readable identification of a subscription. The name must be unique in the database. |
 | **Filter** | `Expression<Func<T, bool>>` | Lambda describing filter logic for the subscription. Will be translated to a javascript function.
-| **Projection** | `Expression<Func<T, object>>` | Lambda describing the projection of returned documents. |Will be translated to a javascript function
+| **Projection** | `Expression<Func<T, object>>` | Lambda describing the projection of returned documents. Will be translated to a javascript function. |
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown:62]: can you change 'javascript' -> 'JavaScript' ?

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown [hunk 3]
```diff
@@ -50,18 +52,51 @@ Non generic version of the class, relies on user's full knowledge of the RQL que
...
+
+| Return value | Description |
+| ------------- | ----- |
+| `string` | The updated data subscription's name. |
+
+{PANEL/}
+
+{PANEL: SubscriptionUpdateOptions}
+
+Inherits from `SubscriptionCreationOptions` and has all the same fields (see [above](../../../client-api/data-subscriptions/creation/api-overview#subscriptioncreationoptions)) plus two.
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown:89]: plus two -> with additional ones described below

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown [hunk 4]
```diff
@@ -50,18 +52,51 @@ Non generic version of the class, relies on user's full knowledge of the RQL que
...
+
+{PANEL: SubscriptionUpdateOptions}
+
+Inherits from `SubscriptionCreationOptions` and has all the same fields (see [above](../../../client-api/data-subscriptions/creation/api-overview#subscriptioncreationoptions)) plus two.  
+
+{CODE sub_update_options@ClientApi\DataSubscriptions\DataSubscriptions.cs /}
+
+| Parameter | Type | Description |
+| - | - | - |
+| **Id** | `long?` | Unique server-side ID of the data subscription, see description of the [subscription state object](../../../client-api/data-subscriptions/advanced-topics/maintenance-operations#getting-subscription-status). `Id` can be used instead of the subscription update options `Name` field, and takes precedence over it. This allows you to change the subscription's name: submit a subscription's id, and submit a different name in the `Name` field. |
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown:95]: subscription's id -> subscription's ID

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown [hunk 5]
```diff
@@ -89,9 +88,16 @@ Please see the [page](../../../client-api/data-subscriptions/advanced-topics/sub
 
 {PANEL/}
 
+{PANEL:Update existing subscription}
+Here we update the filter query of an existing data subscription named "my subscription".  
+{CODE:csharp update_subscription_example_0@ClientApi\DataSubscriptions\DataSubscriptions.cs /}
 
-
-
+In addition to names, subscriptions also have a **subscription id** on the server side. The
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown:95]: id -> ID

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown [hunk 6]
```diff
@@ -89,9 +88,16 @@ Please see the [page](../../../client-api/data-subscriptions/advanced-topics/sub
...
 {PANEL/}
 
+{PANEL:Update existing subscription}
+Here we update the filter query of an existing data subscription named "my subscription".  
+{CODE:csharp update_subscription_example_0@ClientApi\DataSubscriptions\DataSubscriptions.cs /}
 
-
-
+In addition to names, subscriptions also have a **subscription id** on the server side. The 
+id can be used to identify the subscription instead of using its name. This allows use to change
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown:96]: id -> ID

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown [hunk 7]
```diff
@@ -89,9 +88,16 @@ Please see the [page](../../../client-api/data-subscriptions/advanced-topics/sub
...
 
+{PANEL:Update existing subscription}
+Here we update the filter query of an existing data subscription named "my subscription".  
+{CODE:csharp update_subscription_example_0@ClientApi\DataSubscriptions\DataSubscriptions.cs /}
 
-
-
+In addition to names, subscriptions also have a **subscription id** on the server side. The 
+id can be used to identify the subscription instead of using its name. This allows use to change 
+an existing subscription's name by specifying the subscription with the id, and submitting
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown:97]: id -> ID

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.dotnet.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown (hunk 2 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown (hunk 3 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown (hunk 4 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown (hunk 5 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown (hunk 6 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown (hunk 7 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/examples.dotnet.markdown]: fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+| **Projection** | `Expression<Func<T, object>>` | Lambda describing the projection of returned documents. Will be translated to a JavaScript function. |
+| **Includes** | `Action<ISubscriptionIncludeBuilder<T>>` | Action with an [ISubscriptionIncludeBuilder](../../../client-api/data-subscriptions/creation/examples#create-subscription-with-include-statement) parameter that allows you to define includes for the subscription. |
+| **ChangeVector** | `string` | Allows to define a change vector, from which the subscription will start processing. It might be useful for ad-hoc processes that need to process only recent changes in data, for that specific use, the field may receive a special value: "LastDocument", that will take the latest change vector in the machine. |
+| **MentorNode** | `string` | Allows to define a specific node in the cluster that we want to treat the subscription. That's useful in cases when one server is preffered over other, either because of stronger hardware or closer geographic proximity to clients etc. |
+
+{PANEL/}
+
+{PANEL: Update Subscription}
+
+Modifies an existing data subscription. Method is accessible at `DocumentStore.Subscriptions.Update()`.
```
Comment (garayx) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown:71]:
```suggestion
Modifies an existing data subscription. Method is accessible at `DocumentStore.Subscriptions`.
```

Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/data-subscriptions/creation/api-overview.dotnet.markdown]: Fixed

**Review by garayx — APPROVED** [review]

**Comment by ppekrol** [comment]
@morhilai can you rebase and fix the conflicts?

**Pull request closed without merging**